src/components/ConnectionPage/Dashboard.tsx

```diff
@@ -0,0 +1,178 @@
+      setMyConnections(mappedConnections);
+    } catch (error) {
+      console.error('Error fetching connections:', error);
+    }
+  };
+
+  const handleSendConnectionRequest = async (recipientId: number) => {
+    try {
+      if (recipientId !== userId) {
+        await sendConnectionRequest(userId, recipientId);
+        const recipientName = searchResults.find(user => user.id === recipientId)?.username;
+        toast.success(`You have sent a connection request to ${recipientName}!`);
+        await fetchPendingRequests();
+      } else {
+        toast.error("You cannot send a connection request to yourself.");
+      }
+    } catch (error) {
+      toast.error('Error sending connection request.');
+      console.error('Error sending connection request:', error);
+    }
+  };
+
+  const handleAcceptRequest = async (requestId?: number) => {
+    if (requestId) {
+      try {
+        await acceptConnectionRequest(requestId);
+        toast.success('Connection request accepted!');
+        await fetchPendingRequests();
+        await fetchMyConnections();
+      } catch (error) {
+        toast.error('Error accepting request.');
+        console.error('Error accepting request:', error);
+      }
+    }
+  };
+
+  const handleRejectRequest = async (requestId?: number) => {
+    if (requestId) {
+      try {
+        await rejectConnectionRequest(requestId);
+        toast.info('Connection request rejected.');
+        await fetchPendingRequests();
+      } catch (error) {
+        toast.error('Error rejecting request.');
+        console.error('Error rejecting request:', error);
+      }
+    }
+  };
+
+  const handleRemoveConnection = async (connectionId?: number) => {
+    if (connectionId) {
+      try {
+        await removeConnection(connectionId);
+        toast.info('Connection removed.');
+        await fetchMyConnections();
+      } catch (error) {
+        toast.error('Error removing connection.');
+        console.error('Error removing connection:', error);
+      }
+    }
+  };
+
+  return (
+    <div className="dashboard">
+      <ToastContainer />
+      <Header />
+      <main className="main-content">
+        <SearchBar
+          searchQuery={searchQuery}
+          handleSearchChange={handleSearchChange}
+          searchResults={searchResults}
+          handleSendConnectionRequest={handleSendConnectionRequest}
+        />
+        <PendingRequests
+          pendingRequests={pendingRequests}
+          handleAcceptRequest={handleAcceptRequest}
+          handleRejectRequest={handleRejectRequest}
+        />
+        <MyConnections
+          myConnections={myConnections}
+          handleRemoveConnection={handleRemoveConnection}
+        />
+      </main>
+      <footer className="footer">
+        <p>&copy; 2024 Revature. All rights reserved.</p>
+        <nav className="footer-nav">
+          <a href="/about">About</a>
+          <a href="/privacy">Privacy Policy</a>
+          <a href="/contact">Contact Us</a>
```

Comment:
Use `<Link />` tags here so that we save the browser memory. They are a apart of `react-router-dom` components.